src/dict.c

```diff
@@ -1275,7 +1274,7 @@ dict_count(dict_T *d, typval_T *needle, int ic)
 	if (!HASHITEM_EMPTY(hi))
 	{
 	    --todo;
-	    if (tv_equal(&HI2DI(hi)->di_tv, needle, ic, FALSE))
```

Comment:
For example, `dict_count()` is the first time to call `tv_equal()`, so it pass FALSE to `tv_equal()`